src/generators/app.ts

```diff
@@ -1,708 +1,432 @@
+const rmrf = isWindows ? 'rimraf' : 'rm -rf';
+const rmf = isWindows ? 'rimraf' : 'rm -f';
+
+let successfullyInstalledDeps = false;
+
+interface IPjson {
+  name?: string;
+  description?: string;
+  author?: string;
+  homepage?: string;
+  bugs?: string;
+  version?: string;
+  license?: string;
+  files?: string[];
+  repository?: string;
+  dependencies?: AnyJson;
+  devDependencies?: AnyJson;
+  oclif?: {
+    commands?: string;
+    bin?: string;
+    dirname?: string;
+    topics?: AnyJson;
+    plugins: string[];
+    devPlugins: string[];
+  };
+  scripts?: {
+    test?: AnyJson;
+    prepack?: AnyJson;
+    build?: AnyJson;
+    lint?: AnyJson;
+    posttest?: AnyJson;
+    postpack?: AnyJson;
+    version?: AnyJson;
+  };
+  keywords?: string[];
+  engines?: JsonMap;
+}
+
+interface IAnswers {
+  name: string;
+  description: string;
+  version: string;
+  github?: { repo: string; user: string };
+  author: string;
+  license: string;
+}
 
 class App extends Generator {
-  options: {
-    defaults?: boolean
-    mocha: boolean
-    circleci: boolean
-    appveyor: boolean
-    typescript: boolean
-    tslint: boolean
-    eslint: boolean
-    yarn: boolean
-    travisci: boolean
-  }
+  public options: {
+    defaults?: boolean;
+    skipInstall?: boolean;
+  };
 
-  args!: {[k: string]: string}
-  type: 'single' | 'multi' | 'plugin' | 'base' | 'sfdx-plugin'
-  path: string
-  pjson: any
-  githubUser: string | undefined
-  answers!: {
-    name: string
-    bin: string
-    description: string
-    version: string
-    github: {repo: string, user: string}
-    author: string
-    files: string
-    license: string
-    pkg: string
-    typescript: boolean
-    tslint: boolean
-    eslint: boolean
-    mocha: boolean
-    ci: {
-      circleci: boolean
-      appveyor: boolean
-      travisci: boolean
-    }
-  }
+  public repository?: string;
 
-  mocha!: boolean
-  circleci!: boolean
-  appveyor!: boolean
-  ts!: boolean
-  tslint!: boolean
-  eslint!: boolean
-  yarn!: boolean
-  travisci!: boolean
-  get _ext() {
-    return this.ts ? 'ts' : 'js'
-  }
+  private path: string;
+  private pjson: IPjson;
+  private githubUser: string | undefined;
+  private answers!: IAnswers;
 
-  get _bin() {
+  // eslint-disable-next-line no-underscore-dangle
+  private get _bin(): string {
     // eslint-disable-next-line no-mixed-operators
-    let bin = this.pjson.oclif && (this.pjson.oclif.bin || this.pjson.oclif.dirname) || this.pjson.name
-    if (bin.includes('/')) bin = bin.split('/').pop()
-    return bin
-  }
 
-  repository?: string
+    let bin = (this.pjson.oclif && (this.pjson.oclif.bin || this.pjson.oclif.dirname)) || this.pjson.name;
+    if (bin.includes('/')) bin = bin.split('/').pop();
+    return bin;
+  }
 
-  constructor(args: any, opts: any) {
-    super(args, opts)
+  public constructor(args: string | string[], opts: Options) {
+    super(args, opts, {
+      customInstallTask: true,
```

Comment:
See @gbockus-sf  findings: https://github.com/forcedotcom/salesforcedx-templates/pull/476#discussion_r845740693

So, `yeoman/environment` tries to detect which package manager to use and if it detect changes to the pjson (only detect those if it's modified using `this.packageJson.*` inside the generator class) it tries to install deps.

In our case we do want yeoman to install deps, but we do that manually by implementing the `install` method: 
https://github.com/forcedotcom/sfdx-plugin-generate/blob/b9b78243c8de65191316902656ed413311a11c1d/src/generators/app.ts#L330

without setting `customInstallTask` to true yeoman will try re-install deps again:

![Screen Shot 2022-05-03 at 16 33 29](https://user-images.githubusercontent.com/6853656/166552599-8eccba4a-2ff2-440b-b39c-bd91a838b12c.png)